mm/2s2h/Rando/ActorBehavior/Player.cpp

```diff
@@ -30,4 +39,20 @@ void Rando::ActorBehavior::InitPlayerBehavior() {
             RespawnOnWaterTouch(GET_PLAYER(gPlayState));
         }
     });
+    COND_VB_SHOULD(VB_PLAY_OCARINA_NOTE, IS_RANDO, {
+        u8* sCurOcarinaButtonIndex = va_arg(args, u8*);
+        u8* sCurOcarinaPitch = va_arg(args, u8*);
+        bool canPlayNote = true;
+
+        auto findNote = ocarinaButtonToFlagMap.find(static_cast<OcarinaButtonIndex>(*sCurOcarinaButtonIndex));
+        if (findNote != ocarinaButtonToFlagMap.end()) {
+            canPlayNote = Flags_GetRandoInf(findNote->second);
+        }
+
+        if (!canPlayNote) {
```

Comment:
Is this map really needed? You should be able to just do some arithmetic:

```suggestion
        if(!Flags_GetRandoInf(((*sCurOcarinaButtonIndex) - OCARINA_BTN_A) + RANDO_INF_OBTAINED_OCARINA_BUTTON_A)) {
```